### PR TITLE
read_json `engine` argument integration

### LINKED ISSUE #48893: ENH: Add `engine` keyword to `read_json` to enable reading from pyarrow
pyarrow has a `read_json` function that could be used as an alternative parser for `pd.read_json` https://arrow.apache.org/docs/python/generated/pyarrow.json.read_json.html#pyarrow.json.read_json

Like we have for `read_csv` and `read_parquet`, I would like to propose a `engine` keyword argument to allow users to pick the parsing backend `engine="ujson"|"pyarrow"`

One change compared to `read_csv` and `read_parquet` with `engine="pyarrow"` I would like to propose would be to return `ArrowExtensionArray`s instead of converting the result to numpy dtypes such that the `pyarrow.Table` returned by `read_json` still propagates pyarrow objects underneath.

Thoughts?

### PR DESCRIPTION
**Pull request by abkosar**
Creating the PR so we can start the discussion. It's not complete yet I will keep updating the PR; however, any feedback of course is appreciated. 

My main question is, there is an `arrow_parser_wrapper.py` file has a `ArrowParserWrapper` class which mainly serves the `engine` argument of the `read_csv` method. I though of extending that classes functionality so it can serve both `read_json` and `read_csv` but wanted to make get your opinion about it.

I will also add tests but I am still reading the tests part from the contributing guidelines.

- [ ] closes #48893 (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/io/json/_json.py [hunk 1]
```diff
@@ -780,6 +792,7 @@ def __init__(
         precise_float: bool,
         date_unit,
         encoding,
+        engine,
```
Comment (mroeschke) [inline pandas/io/json/_json.py:795]: This keyword will need to go at the end of the function to accommodate people passing positional args.

File: pandas/io/json/_json.py [hunk 2]
```diff
@@ -607,6 +615,9 @@ def read_json(
 
         .. versionadded:: 1.3.0
 
+    engine : {{'ujson', 'pyarrow'}}
```
Comment (mroeschke) [inline pandas/io/json/_json.py:618]: This should default to `"ujson"`

**Comment by abkosar** [comment]
Yeah that makes sense. Sounds good then, will do that. Thanks for the feedback!

File: pandas/io/json/_json.py (hunk 2 above)
Comment (abkosar) [reply pandas/io/json/_json.py:618]: I actually have a question about this. I thought that the `engine` argument will be optional. Or does the `read_json` method uses ujson currently and we are just adding another option?

Comment (mroeschke) [reply pandas/io/json/_json.py:618]: The existing parsing logic is (vendored) ujson code, so that's why the "default" should be "ujson".

Comment (abkosar) [reply pandas/io/json/_json.py:618]: Yeah after I made that comment that's what I figured.

**Comment by abkosar** [comment]
I just wanted to give an update. I didn't forget about the issue, I'm working on it:) After days of staring between source code and me finally I was able to implement a logic the past two days and wrote a test for it. I have been trying to get the docker interpreter working so I can run the tests. After that I'll update the PR.

**Comment by abkosar** [comment]
@mroeschke I have to close this PR since I messed up something in git history. I have to create a new PR from main branch. Is there a closing process, any commands I should run?

**Comment by abkosar** [comment]
Opened #49249 instead of this PR.

**Comment by mroeschke** [comment]
Closing in favor of https://github.com/pandas-dev/pandas/pull/49249

**Pull request closed without merging**